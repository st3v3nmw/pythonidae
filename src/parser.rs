use insta::assert_debug_snapshot;

lalrpop_mod!(pub grammar);

#[test]
fn raw_expressions() {
    assert_debug_snapshot!(grammar::ProgramParser::new().parse("42"));
    assert_debug_snapshot!(grammar::ProgramParser::new().parse("(42)"));
    assert_debug_snapshot!(grammar::ProgramParser::new().parse("((((42))))"));
    assert_debug_snapshot!(grammar::ProgramParser::new().parse("((42)"));
    assert_debug_snapshot!(grammar::ProgramParser::new().parse("2 * 4 - 3"));
    assert_debug_snapshot!(grammar::ProgramParser::new().parse("(2 * 4) - 3"));
    assert_debug_snapshot!(grammar::ProgramParser::new().parse("(2-4) /3"));
}

#[test]
fn assignments() {
    assert_debug_snapshot!(grammar::ProgramParser::new().parse(
        r#"
            foo=2 * 4 - 3
            bar = (2-4) /3
        "#
    ));
}
