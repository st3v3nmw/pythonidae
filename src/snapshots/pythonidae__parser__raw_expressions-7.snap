---
source: src/parser.rs
expression: "grammar::ProgramParser::new().parse(\"(2-4) /3\")"
---
Ok(
    [
        RawExpr(
            Op(
                Op(
                    Number(
                        2,
                    ),
                    Sub,
                    Number(
                        4,
                    ),
                ),
                Div,
                Number(
                    3,
                ),
            ),
        ),
    ],
)
